Add ecosystem schema sync MVP

MythicManiac opened the pull request:
Implement a minimum viable version of a simple schema sync feature, which imports a games.json from the configured endpoint and automatically provisions communities for games with the configurations for doing so.

Importantly also include a feature that can be used to opt-out communities from this automation, and enable that by default for all pre-existing communities.

Refs TS-1429

anttimaki commented on django/thunderstore/schema/sync.py:
```diff
@@ -0,0 +1,84 @@
+
+    config = game.thunderstore
+
+    if not (community := Community.objects.filter(identifier=identifier).first()):
```
Assignment expressions are cool and all, but would Django's `get_or_create` or `update_or_create` be more idiomatic? This comment also applies for `PackageCategory` and `PackageListingSection` stuff below.

MythicManiac replied:
The problem with those is that you might want to match the object with different criteria than you'd want to update or create it with. Maybe `update_or_create` is flexible enough to handle everything correctly, but I seem to remember that it wasn't.

That said, this area is a good spot for improvements later on. Really what we should use is an upsert query (bulk insert that on conflict updates specific fields instead), but Django doesn't have built in support for it and I didn't want to add needless complexity here yet.

Very valid suggestions though and I agree with the sentiment

anttimaki commented on django/thunderstore/schema/sync.py:
```diff
@@ -0,0 +1,84 @@
+    if community.block_auto_updates:
+        return
+
+    community.slogan = f"The {game.meta.displayName} Mod Database"
```
Is it intentional the slogan and description are always overwritten by hardcoded values? Should this be done on L22 instead when a new community is created?

MythicManiac replied:
Really these fields should be removed from the Community model entirely as all of them follow this same pattern currently. As such I figured it's better to force them to remain in line with that pattern than it is to allow overwrites to it in the admin.

anttimaki reviewed (approved):
Two comments but no blockers.

MythicManiac commented:
Rebased to latest master and changed the schema URL to be treated as an absolute URL rather than something we need to append to (as it matches the design I have in mind better).
